tcms/testcases/tests/test_views.py

```diff
@@ -338,6 +338,15 @@ def test_show_clone_page_with_selected_cases(self):
             self.assertContains(response,
                                 "TC-%d: %s" % (case.pk, case.summary))
 
+    def test_user_without_permission_should_not_be_able_to_clone_a_case(self):
+        remove_perm_from_user(self.tester, 'testcases.add_testcase')
+        url = reverse('testcases-clone') + quote('?case=' + str(self.case_1.pk))
```

Comment:
You don't need the `quote()` function here, do you ?

Reply:
`'?case=' + str(self.case_1.pk)` -> `"?case=%d" % self.case_1.pk`

Reply:
Actually you don't need to add the query params to the URL at all because `self.get()` below does this for you, doesn't it ?

Reply:
It does, but this is for the expected result, the url variable is for the actual result of the assert.